app/controllers/api/v2/search_controller.rb

```diff
@@ -12,6 +12,7 @@ class Api::V2::SearchController < Api::BaseController
     before_action :query_pagination_error, if: :pagination_requested?
     before_action :remote_resolve_error, if: :remote_resolve_requested?
   end
+  before_action :query_offset_error, if: [:pagination_requested?, :invalid_offset_value?]
```

Comment:
I assume from your code that the handling of arrays in a `if:` is a `and`, but I had not seen this pattern before, and I can't find the implementation nor documentation, so maybe it would make more sense to have the full check in a single method.

Reply:
Moved the before_action definition to base, left the call to use it in search, added a spec for `limit` (previously was only `offset`, and removed the chained-symbol-array `if` options.